ForumProject/users/urls.py

```diff
@@ -1,9 +1,11 @@
 from django.urls import path
+from rest_framework_simplejwt.views import TokenObtainPairView, TokenRefreshView
+
 from . import views
 
 app_name = 'users'
 
 urlpatterns = [
-    
-    
-]
\ No newline at end of file
+    path('token/', TokenObtainPairView.as_view(), name='token_obtain_pair'),
+    path('token/refresh/', TokenRefreshView.as_view(), name='token_refresh'),
```

Comment:
Consider implementing rate limiting on token-related endpoints to prevent abuse, as these endpoints can be particularly sensitive to brute-force attacks
https://www.django-rest-framework.org/api-guide/throttling/
```
# In settings.py

REST_FRAMEWORK = {
    'DEFAULT_THROTTLE_CLASSES': [
        'rest_framework.throttling.AnonRateThrottle',
        'rest_framework.throttling.UserRateThrottle'
    ],
    'DEFAULT_THROTTLE_RATES': {
        'anon': '100/day',
        'user': '1000/day',
        'token_obtain': '5/hour',  # Custom scope for obtaining token
        'token_refresh': '10/hour',  # Custom scope for refreshing token
    }
}
```

```
# In users/urls.py

from rest_framework.throttling import ScopedRateThrottle

urlpatterns = [
    path('token/', TokenObtainPairView.as_view(throttle_classes=[ScopedRateThrottle]), name='token_obtain_pair'),
    path('token/refresh/', TokenRefreshView.as_view(throttle_classes=[ScopedRateThrottle]), name='token_refresh'),
]
```

```
# In your users/views.py (or wherever you want to customize the views)

from rest_framework_simplejwt.views import TokenObtainPairView as BaseTokenObtainPairView

class TokenObtainPairView(BaseTokenObtainPairView):
    throttle_scope = 'token_obtain'
```
And similarly for the `TokenRefreshView`

Reply:
Done, thanks